Sources/Private/Experimental/Layers/ShapeLayer.swift

```diff
@@ -83,14 +62,92 @@ final class ShapeLayer: BaseCompositionLayer {
       fatalError("\(Self.self).init(layer:) incorrectly called with \(type(of: layer))")
     }
 
-    shapeLayer = typedLayer.shapeLayer
+    group = typedLayer.group
+    inheritedItems = typedLayer.inheritedItems
     super.init(layer: typedLayer)
   }
 
   // MARK: Private
 
-  private let shapeLayer: ShapeLayerModel
+  private let group: Group
+
+  /// `ShapeItem`s that were listed in the parent's `items: [ShapeItem]` array
+  ///   - This layer's parent is either the root `ShapeLayerModel` or some other `Group`
+  private let inheritedItems: [ShapeItemLayer.Item]
+
+  private func setupLayerHierarchy() {
+    // Groups can contain other groups, so we may have to continue
+    // recursively creating more `GroupLayer`s
+    setupGroups(from: group.items, parentGroup: group)
 
+    let nonGroupItems = group.items
+      .filter { !($0 is Group) }
+      .map { ShapeItemLayer.Item(item: $0, parentGroup: group) }
+      + inheritedItems
+
+    let pathDrawingItemsInGroup = nonGroupItems.filter { $0.item.drawsCGPath }
+    let otherItemsInGroup = nonGroupItems.filter { !$0.item.drawsCGPath }
```

Comment:
Wondering if it might make sense to use:
```swift
Dictionary(grouping: nonGroupItems, by: { $0.item.drawsCGPath })
```
to group them items so we only do one iteration of `nonGroupItems` here. Then we could reference them as:
```swift
let pathDrawingItemsInGroup = grouped[true, default: []] 
let otherItemsInGroup = grouped[false, default: []]
```
Not sure how big `nonGroupItems` could be so it might not be worth it but it's just a thought

Reply:
ohoh good idea! thats a nice simplification